research/workflow-engines/celery-canvas.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Celery Canvas
+authors: lbarcziova
+---
+
+- docs: https://docs.celeryq.dev/en/stable/userguide/canvas.html
+- provides capabilities to build simple workflows
+- it is possible to define e.g. serial execution, in parallel, in group
+
+## Chains
+
+- tasks can be linked together: the linked task is called when the task returns successfully
```

Comment:
Do you know if it's possible to include external events in this? (That's the key feature I am worried about across all the solutions -- how to include result reporting.)

Reply:
not sure how would you expect this to work :thinking: it is needed to create the tasks so that you will can link them together/specify the dependencies

Reply:
My point is that we don't have a workflow where everything happens on our side and the workflow engine "just" decides order and go one-by-one. Each step of our workflow is triggered on our side but needs results from an external service => and I am unsure how these engines can cover this. (The process looks like a workflow for the user, but for us, it's just one task done as a reaction to an external event.)

Reply:
yes, I get that. But I am afraid this "custom" logic needs to be always implemented by us, no tool can do this for us :/ But I will give it more thought.

Reply:
actually, in Airflow there are "Sensors" that could probably help us with that, will add this

Reply:
@lbarcziova , actually, @majamassarini just found that Temporal might have something for that and covered that in her research (#204)